cmd/collector/app/flags/flags.go

```diff
@@ -258,32 +232,32 @@ func (cOpts *CollectorOptions) InitFromViper(v *viper.Viper, logger *zap.Logger)
 	cOpts.QueueSize = v.GetUint(flagQueueSize)
 	cOpts.DynQueueSizeMemory = v.GetUint(flagDynQueueSizeMemory) * 1024 * 1024 // we receive in MiB and store in bytes
 	cOpts.SpanSizeMetricsEnabled = v.GetBool(flagSpanSizeMetricsEnabled)
+	cOpts.Tenancy = tenancy.InitFromViper(v)
 
-	if err := cOpts.HTTP.initFromViper(v, logger, httpServerFlagsCfg); err != nil {
+	if err := initHTTPFromViper(v, logger, &cOpts.HTTP, nil, httpServerFlagsCfg); err != nil {
```

Comment:
can we just add cors options for legacy http as well and pass cosrCfg here? We can probably just incorporate corsCfg into ServerFlagsCfg